subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -148,10 +148,20 @@ static bool validate_firmware(u32_t fw_dst_address, u32_t fw_src_address,
 
 	for (u32_t key_data_idx = 0; key_data_idx < num_public_keys;
 			key_data_idx++) {
-		if (public_key_data_read(key_data_idx, &key_data[0],
-				CONFIG_SB_PUBLIC_KEY_HASH_LEN) < 0) {
-			retval = -EFAULT;
-			break;
+		int retval2 = public_key_data_read(key_data_idx, &key_data[0],
```

Comment:
Since this is not the return value of the function it's a bit misleading calling it `retval2` a naming suggestion could be `valid_key` or something along those lines to increase readability IMO. If you disagree leave it as is.

Reply:
I like `retval` since it is a return value and can contain an errno. How about `key_read_retval`?

Reply:
I like `key_read_retval` at least it is a bit more readable but a bit more verbose too.